engine/src/org/pentaho/di/trans/steps/fileinput/BaseFileInputStep.java

```diff
@@ -427,9 +427,14 @@ protected void fillFileAdditionalFields( D data, FileObject file ) throws FileSy
     data.path = KettleVFS.getFilename( file.getParent() );
     data.hidden = file.isHidden();
     data.extension = file.getName().getExtension();
-    data.lastModificationDateTime = new Date( file.getContent().getLastModifiedTime() );
     data.uriName = file.getName().getURI();
     data.rootUriName = file.getName().getRootURI();
-    data.size = file.getContent().getSize();
+    if ( file.getType().hasContent() ) {
```

Comment:
I am sure this could use a unit (or maybe an intergration) test to prevent a regression in this area.